docs/data/data-grid/column-dimensions/column-dimensions.md

```diff
@@ -88,22 +85,52 @@ In all the cases, the `colDef.minWidth` and `colDef.maxWidth` options will be re
 
 {{"demo": "ColumnAutosizing.js", "disableAd": true, "bg": "inline"}}
 
-:::warning
-The Data Grid can only autosize based on the currently rendered cells.
+:::info
+The `autosizeOptions` only applies to the [column header separator](#column-header-separator) and the `autosizeOnMount` prop.
 
-DOM access is required to accurately calculate dimensions, so unmounted cells (when [virtualization](/x/react-data-grid/virtualization/) is on) cannot be sized. If you need a bigger row sample, [open an issue](https://github.com/mui/mui-x/issues) to discuss it further.
+It **does not** apply to the `autosizeColumns()` method when you call it programmatically, you have to pass the options directly.
 :::
 
-### Autosizing asynchronously
+### Column header separator
+
+Autosizing can be triggered by double-clicking the column header separator.
+
+To disable this behavior, use the `disableAutosize` prop.
+
+```jsx
+<DataGrid {...otherProps} disableAutosize />
+```
+
+Note that for the separator double-click method, the `autosizeOptions.columns` will be replaced by the respective column user double-clicked on.
+
+### Autosizing on mount
+
+To automatically autosize columns when the Data Grid is mounted, use the `autosizeOnMount` prop.
+
+```jsx
+<DataGrid {...otherProps} autosizeOnMount />
+```
+
+### Autosizing programmatically
 
-The `autosizeColumns` method from the `apiRef` can be used as well to adjust the column size on specified events, for example when receiving row data from the server.
+Use `autosizeColumns()` from the `apiRef` to adjust the column size on specified events, for example when receiving row data from the server.
 
 {{"demo": "ColumnAutosizingAsync.js", "disableAd": true, "bg": "inline"}}
 
 :::warning
 This example uses `ReactDOM.flushSync`. If used incorrectly it can hurt the performance of your application. Please refer to the official [React docs](https://react.dev/reference/react-dom/flushSync) for further information.
 :::
 
+### Autosizing virtualized columns
+
+Use `autoSizeColumns({ disableColumnVirtualization: true})` to include columns that are not visible in the DOM due to column virtualization.
```

Comment:
This should work with `autosizeOptions` prop too, right?
```suggestion
Use the option `disableColumnVirtualization: true` to include columns that are not visible in the DOM due to column virtualization.
```

Reply:
It's mentioned in the latest docs too. "To configure autosize for the other two methods, provide the options in the autosizeOptions prop."

https://mui.com/x/react-data-grid/column-dimensions/#autosizing